### PR TITLE
Create controllers to add users and track courses

### PR DESCRIPTION
**Pull request by manishsaha**
## Overview
I added two endpoints to handle user addition as well as course tracking. Making progress!

## Changes Made
I created two new controllers for the functionality described above. I also fixed some minor errors in the course model, which was preventing new tuples from being added properly in our many to many relationship.

## Test Coverage

I used Postman to test both of these routes, both with valid and invalid request bodies.

## Next Steps 
Work on deleting courses from user tracking lists and querying by catalog number to check the current status (open, closed)

### CODE REVIEW

File: src/app/coursegrab/controllers/create_user_controller.py [hunk 1]
```diff
@@ -0,0 +1,25 @@
+from . import *
+
+
+class CreateUserController(AppDevController):
+    def get_path(self):
+        return "/add_user/"
```
Comment (kq-li) [inline src/app/coursegrab/controllers/create_user_controller.py]: prefer `/users/create/`

File: src/app/coursegrab/controllers/track_course_controller.py [hunk 2]
```diff
@@ -0,0 +1,19 @@
+from . import *
+
+
+class TrackCourseController(AppDevController):
+    def get_path(self):
+        return "/track_course/"
```
Comment (kq-li) [inline src/app/coursegrab/controllers/track_course_controller.py]: `/users/track/` or `/courses/track/`

File: src/app/coursegrab/dao/users_dao.py [hunk 3]
```diff
@@ -19,3 +19,19 @@ def create_user(email, first_name, last_name):
     db.session.add(user)
     db.session.commit()
     return True, user
+
+
+def add_course(user_id, catalog_num):
```
Comment (kq-li) [inline src/app/coursegrab/dao/users_dao.py]: `track_course`

File: src/run.py [hunk 4]
```diff
@@ -2,4 +2,4 @@
 from os import environ
 
 if __name__ == "__main__":
-    app.run(host="0.0.0.0", port=environ["PORT"])
+    app.run(host="0.0.0.0", port=environ["PORT"], debug=True)
```
Comment (kq-li) [inline src/run.py]: leave this out

Comment (manishsaha) [reply src/run.py]: this is for hot reloading

Comment (kq-li) [reply src/run.py]: yeah but we don't want this in production

**Pull request merged**